packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js

```diff
@@ -1,24 +1,50 @@
-import * as React from "react";
+import { useEffect, useMemo, useState } from "react";
 import fileSize from "pretty-bytes";
 import { Link } from "gatsby";
+import useSWR from "swr";
+
+import { useUser } from "../../contexts/user";
+import useActivePlan from "../../hooks/useActivePlan";
+import { ContainerLoadingIndicator } from "../LoadingIndicator";
 
 import { GraphBar } from "./GraphBar";
 import { UsageGraph } from "./UsageGraph";
 
-// TODO: get real data
-const useUsageData = () => ({
-  files: {
-    used: 19_521,
-    limit: 20_000,
-  },
-  storage: {
-    used: 23_000_000_000,
-    limit: 1_000_000_000_000,
-  },
-});
+const useUsageData = () => {
+  const { user } = useUser();
+  const { activePlan, error } = useActivePlan(user);
+  const { data: stats, error: statsError } = useSWR("user/stats");
+
+  const [loading, setLoading] = useState(true);
+  const [usage, setUsage] = useState({});
+
+  const hasError = error || statsError;
+  const hasData = activePlan && stats;
+
+  useEffect(() => {
+    if (hasData || hasError) {
+      setLoading(false);
+    }
+
+    if (hasData && !hasError) {
+      setUsage({
+        filesUsed: stats?.numUploads,
+        filesLimit: activePlan?.limits?.maxNumberUploads,
+        storageUsed: stats?.totalUploadsSize,
+        storageLimit: activePlan?.limits?.storageLimit,
+      });
+    }
+  }, [hasData, hasError, stats, activePlan]);
```

Comment:
`nit` sorry for nitting on a change but just for my understand, `stats` and `activeplan` are redundant here right?

Reply:
No :)

* `activePlan` contains information about the active subscription, including bandwidth and storage limits
* `stats` contain the current usage of those limits